publicode/rules/dirigeant.yaml

```diff
@@ -449,7 +449,10 @@ dirigeant . indépendant . assiette des cotisations:
   formule:
     encadrement:
       plancher: 0
-      valeur: revenu professionnel
+      valeur:
+        allègement:
```

Comment:
Les cotisations des contrat madelin sont déductible à l'impôt sur le revenu, mais pas pour le calcul des cotisations sociale.

Ce sont des montant défiscalisés, mais pas désocialisés.

Source : 
> https://www.urssaf.fr/portail/home/independant/mes-cotisations/les-etapes-de-calcul/la-declaration-sociale-des-indep/les-revenus-pris-en-compte-pour.html
> Le revenu soumis aux cotisations obligatoires correspond à celui retenu pour le calcul de l’impôt sur le revenu, sans qu’il soit tenu compte :
- de la déduction des cotisations versées aux régimes facultatifs « loi Madelin » par les assurés y ayant adhéré après le 13 février 1994
- ...

Il faut donc modifier l'assiette de l'impôt sur le revenu, ainsi que le revenu net.

Reply:
🤞je tente le coup: il s'agit donc de modifier `assiette des cotisations`?

Reply:
Si je traduis le texte que tu as envoyé, littéralement ça donne: `assiette des cotisation = revenu professionnel - (deductions madelin)` sachant que `revenu professionnel` est decrit comme le net imposable.

Ceci dit je ne suis pas 100% sûr que le "net imposable" soit la même chose que le "revenu retenu pour le calcul de l’impôt sur le revenu".

Reply:
Je m'auto-réponds: non ce n'est pas ça car (à part le fait que je parle de la même chose que j'ai mis dans la PR déjà) l'UI ne me donne pas le résultat que j'attends du calcul décrit en Français. Donc il y a un truc que j'ai mal compris.

Reply:
Ok en fait si :)

Voilà donc ce que j'ai mis là dans le nouveau commit me semble vraiment être la bonne formule.

Le gros takeaway de mes recherches d'aujourd'hui sur le sujet, c'est qu'il faut partir du principe que l'indépendant qui entre ses données a déjà retiré le total Madelin de son BIC/BNC qu'il entre dans le simulateur. Hope I'm right on this one?

A partir de là, ça veut dire que avant cette PR, il avait une simulation trop généreuse car considérant le Madelin comme simplement soustrait à la fois des cotis soc que des impôts.

Donc la solution est de le réinjecter uniquement dans les cotis soc, ce que j'ai fait là (je pense rigoureusement?): en ajoutant le madelin à l'assiette des cotisations, on est en train de dire qu'elle prend en compte le madelin, mais que le net imposable ne le prend pas en compte.

Reply:
> Le gros takeaway de mes recherches d'aujourd'hui sur le sujet, c'est qu'il faut partir du principe que l'indépendant qui entre ses données a déjà retiré le total Madelin de son BIC/BNC qu'il entre dans le simulateur. Hope I'm right on this one

Effectivement, je partage ton analyse. Les cotisations madelins sont considérées comme des charges, et donc retiré de la rémunération totale brut.

> Donc la solution est de le réinjecter uniquement dans les cotis soc, ce que j'ai fait là (je pense rigoureusement?): en ajoutant le madelin à l'assiette des cotisations, on est en train de dire qu'elle prend en compte le madelin, mais que le net imposable ne le prend pas en compte.

Ca me semble être effectivement la bonne approche :+1: